src/overlays/actors/ovl_En_Gs/z_en_gs.c

```diff
@@ -21,35 +37,604 @@ const ActorInit En_Gs_InitVars = {
+                            0, 0, 0, 2);
+                        Audio_PlayActorSound2(&this->actor, NA_SE_EV_BUTTERFRY_TO_FAIRY);
+                    } else {
+                        if (globalCtx->msgCtx.unk_E3F2 == 11) {
+                            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ELF, 
+                                this->actor.posRot.pos.x, 
+                                this->actor.posRot.pos.y + 40.0f, 
+                                this->actor.posRot.pos.z, 
+                                0, 0, 0, 7);
+                            Audio_PlayActorSound2(&this->actor, NA_SE_EV_BUTTERFRY_TO_FAIRY);
+                        }
+                    }
+                    this->unk_19D = 0;
+                    Flags_SetSwitch(globalCtx, (this->actor.params >> 8) & 0x3F);
+                    return;
+                }
+                if (globalCtx->msgCtx.unk_E3EE == 1) {
+                    player->stateFlags2 |= 0x800000;
+                }
+            }
+        }
+    }
+}
+
+void func_80A4E648(EnGs *this, GlobalContext *globalCtx) {
+    s16 sp26;
+    s16 sp24;
+
+    if (this->unk_19C == 1) {
+        func_8010B720(globalCtx, this->actor.textId);
+        this->unk_19C = 2;
+        return;
+    }
+    if (this->unk_19C == 2) {
+        this->unk_19C = func_80A4E3EC(this, globalCtx);
+        return;
+    }
+    if (func_8002F194(&this->actor, globalCtx)) {
+        this->unk_19C = 2;
+        return;
+    }
+    func_8002F374(globalCtx, &this->actor, &sp26, &sp24);
+
+    if (((sp26 >= 0) && (sp26 < 0x141)) && 
+        ((sp24 >= 0) && (sp24 <  0xF1)) &&
+         (this->unk_19C != 3)) {
+            if (func_8002F2CC(&this->actor, globalCtx, 40.0f) == 1) {
+                if (func_8008F080(globalCtx) == 8) {
+                    this->actor.textId = 0x2054U;
+                } else {
+                    this->actor.textId = 0x2053U;
+                }
+            }
+        }
+}
+
+f32 func_80A4E754(EnGs* this, GlobalContext* globalCtx, f32* arg2, f32* arg3, u16* arg4, f32 arg5, f32 arg6, f32 arg7, s32 arg8, s32 arg9) {
+    f32 sp2C;
+
+    sp2C = Math_SmoothScaleMaxMinF(arg2, *arg3, arg5, arg6, arg7);
+    if (arg9 == 0) {
+        sp2C = Math_SmoothScaleMaxMinF(arg2, *arg3, arg5, arg6, arg7);
+        this->unk_1B4[0].x = (sinf((((*arg4 % arg8) * (1.0f / arg8)) * 360.0f) * 0.017453292f) * *arg2) + 1.0f;
+        this->unk_1B4[0].y = 1.0f - (sinf((((*arg4 % arg8) * (1.0f / arg8)) * 360.0f) * 0.017453292f) * *arg2);
+        *arg4 = *arg4 + 1;
+    }
+    return sp2C;
+}
+
+void func_80A4E910(EnGs* this, GlobalContext* globalCtx) {
+    if (this->unk_19F == 0) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_STALKID_ATTACK);
+        this->unk_200 = 0;
+        this->unk_19F = 1;
+        this->unk_1E8 = 0.5f;
+        this->unk_1EC = 0.0f;
+    } else {
+        if (this->unk_19F == 1) {
+            if (func_80A4E754(this, globalCtx, &this->unk_1E8, &this->unk_1EC, 
```

Comment:
This condition can be &&d with the preceeding one I'd think, and this can probably be made into an else if on the same line